### PR TITLE
Scale timeouts for bulk operations based on the number of objects

### PR DESCRIPTION
**Pull request by dshammond**
Scale up bulk-request timeouts based on the number of objects being requested,.

### CODE REVIEW

File: lib/src/RedisRemoteSaiInterface.cpp [hunk 1]
```diff
@@ -2097,3 +2106,20 @@ void RedisRemoteSaiInterface::refreshTableDump()
...
+// Scale a channel timeout interval up based on the object count.
+unsigned int RedisRemoteSaiInterface::scale_timeout_interval(_In_ uint32_t object_count)
+{
+    SWSS_LOG_ENTER();
+
+    // We return a value which is the greater of:
+    // - The default 60s timeout interval
+    // - 50ms per object
+    unsigned int scaled_wait = (unsigned int)object_count * PER_OBJ_TIMEOUT;
+    if (scaled_wait > DEFAULT_GETRESPONSE_TIMEOUT_MS)
```
Comment (abanu-ms) [inline lib/src/RedisRemoteSaiInterface.cpp:2119]: You could return `std::max(object_count * PER_OBJ_TIMEOUT, DEFAULT_GETRESPONSE_TIMEOUT_MS);`

File: lib/inc/Channel.h [hunk 2]
```diff
@@ -46,6 +51,11 @@ namespace sairedis
 
             virtual sai_status_t wait(
                     _In_ const std::string& command,
+                    _Out_ swss::KeyOpFieldsValuesTuple& kco) { return wait(command, DEFAULT_GETRESPONSE_TIMEOUT_MS, kco); }
```
Comment (kcudnik) [inline lib/inc/Channel.h:54]: dont put code in the header if possible

File: lib/inc/RedisRemoteSaiInterface.h [hunk 3]
```diff
@@ -438,6 +438,9 @@ namespace sairedis
 
             void refreshTableDump();
 
+            // Scale a channel timeout interval up based on the object count.
+            static unsigned int scale_timeout_interval(_In_ uint32_t object_count);
```
Comment (kcudnik) [inline lib/inc/RedisRemoteSaiInterface.h:442]: each param should have its own line

File: lib/src/RedisChannel.cpp [hunk 4]
```diff
@@ -9,11 +9,6 @@
 
 using namespace sairedis;
 
-/**
- * @brief Get response timeout in milliseconds.
- */
-#define REDIS_ASIC_STATE_COMMAND_GETRESPONSE_TIMEOUT_MS (60*1000)
```
Comment (kcudnik) [inline lib/src/RedisChannel.cpp:15]: you have not the latest master, this define was changed and moved to sairedis.h

File: lib/src/RedisRemoteSaiInterface.cpp [hunk 5]
```diff
@@ -2097,3 +2106,20 @@ void RedisRemoteSaiInterface::refreshTableDump()
                 kvp.second.size());
     }
 }
+
+// Scale a channel timeout interval up based on the object count.
+unsigned int RedisRemoteSaiInterface::scale_timeout_interval(_In_ uint32_t object_count)
```
Comment (kcudnik) [inline lib/src/RedisRemoteSaiInterface.cpp:2111]: each param per line, please use uint32_t for return type

File: lib/src/RedisRemoteSaiInterface.cpp (hunk 1 above)
Comment (kcudnik) [inline lib/src/RedisRemoteSaiInterface.cpp:2119]: empty lines before if

**Pull request merged**